src/composeApolloProvider.tsx

```diff
@@ -0,0 +1,72 @@
+/// <reference path="../typings/main.d.ts" />
+/* tslint:disable:no-unused-variable */
+import * as React from 'react';
+/* tslint:enable:no-unused-variable */
+import {
+    Component,
+    PropTypes,
+} from 'react';
+
+import {
+    Store,
+} from 'redux';
+
+
+import ApolloClient from 'apollo-client';
+
+declare interface ProviderProps {
+    store?: Store<any>;
+    client: ApolloClient;
+}
+
+export default function composeApolloProvider<T>(Provider: React.ComponentClass<any>): any {
```

Comment:
Actually I think now that we are actually using Redux under the hood for the provider, we should just recommend people use this directly. Now that my feet are more wet into Apollo, I think using the ApolloProvider directly with this implementation makes more sense, than composing it yourself in app.